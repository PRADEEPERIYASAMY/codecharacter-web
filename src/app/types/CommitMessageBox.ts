export interface ElementOwnProps {
  isCommitMessageBoxOpen: boolean;
  commitMessage: string;
  handleCommit: () => void;
  updateCommitMessage: (commitMessage: string) => void;
}

export type Props = ElementOwnProps;
