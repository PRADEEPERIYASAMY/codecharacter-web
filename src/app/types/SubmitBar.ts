export interface State {
  commitMessage: string;
  isCommitMessageBoxOpen: boolean;
}

export interface DispatchProps {
  saveCode: () => void;
  commit: (commitMessage: string) => void;
}

export type Props = {} & DispatchProps;
