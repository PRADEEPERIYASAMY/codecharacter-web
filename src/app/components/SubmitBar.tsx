import { faCloud, faCodeBranch, faPlay } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as styles from 'app/styles/SubmitBar.module.css';
import * as SubmitBarInterfaces from 'app/types/SubmitBar';
import classnames from 'classnames';
import * as React from 'react';

export class SubmitBar extends React.Component<SubmitBarInterfaces.Props, {}> {
  public render() {
    return (
      <div className={classnames(styles.SubmitBar)}>
        <button className={classnames(styles.customBtn)}>
          <div>
            <span className={classnames(styles.icon)}>
              <FontAwesomeIcon icon={faPlay} />
            </span>
            <span>RUN</span>
          </div>
        </button>
        <button className={classnames(styles.customBtn)} onClick={this.props.saveCode}>
          <span className={classnames(styles.icon)}>
            <FontAwesomeIcon icon={faCloud} />
          </span>
          <span>SAVE</span>
        </button>
        <button
          className={classnames(styles.customBtn)}
          onClick={() => this.props.commitCode('Commit Message')}
        >
          <span className={classnames(styles.icon)}>
            <FontAwesomeIcon icon={faCodeBranch} />
          </span>
          <span>COMMIT</span>
        </button>
      </div>
    );
  }
}
