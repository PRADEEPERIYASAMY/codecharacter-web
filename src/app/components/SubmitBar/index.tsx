import { faCloud, faCodeBranch, faPlay } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { CommitMessageBox } from 'app/components/SubmitBar/CommitMessageBox';
import * as styles from 'app/styles/SubmitBar.module.css';
import * as SubmitBarInterfaces from 'app/types/SubmitBar';
import classnames from 'classnames';
import * as React from 'react';

export class SubmitBar extends React.Component<
  SubmitBarInterfaces.Props,
  SubmitBarInterfaces.State
> {
  constructor(props: SubmitBarInterfaces.Props) {
    super(props);
    this.state = {
      commitMessage: '',
      isCommitMessageBoxOpen: false,
    };
  }

  public render() {
    const { saveCode } = this.props;
    const { commitMessage, isCommitMessageBoxOpen } = this.state;
    return (
      <div
        className={classnames(styles.SubmitBar, {
          [`${styles.hideCommitBox}`]: !isCommitMessageBoxOpen,
        })}
      >
        <button className={classnames(styles.customBtn)}>
          <span className={classnames(styles.icon)}>
            <FontAwesomeIcon icon={faPlay} />
          </span>
          <span>RUN</span>
        </button>
        <button className={classnames(styles.customBtn)} onClick={saveCode}>
          <span className={classnames(styles.icon)}>
            <FontAwesomeIcon icon={faCloud} />
          </span>
          <span>SAVE</span>
        </button>
        <button
          className={classnames(styles.customBtn)}
          onClick={() => this.toggleCommitMessageBox(!isCommitMessageBoxOpen)}
        >
          <span className={classnames(styles.icon)}>
            <FontAwesomeIcon icon={faCodeBranch} />
          </span>
          <span>COMMIT</span>
        </button>
        <CommitMessageBox
          commitMessage={commitMessage}
          isCommitMessageBoxOpen={isCommitMessageBoxOpen}
          handleCommit={this.handleCommit}
          updateCommitMessage={this.updateCommitMessage}
        />
      </div>
    );
  }

  private toggleCommitMessageBox = (isCommitMessageBoxOpen: boolean) => {
    this.setState({
      isCommitMessageBoxOpen,
    });
  };

  private updateCommitMessage = (commitMessage: string) => {
    this.setState({
      commitMessage,
    });
  };

  private handleCommit = async () => {
    const { commitMessage } = this.state;
    const { commit, getCommitLog } = this.props;
    await commit(commitMessage);
    await this.toggleCommitMessageBox(false);
    await getCommitLog();
  };
}
