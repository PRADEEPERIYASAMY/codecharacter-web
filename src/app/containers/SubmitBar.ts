import { CodeActions } from 'app/actions';
import { SubmitBar } from 'app/components/SubmitBar';
import { RootState } from 'app/reducers';
import * as SubmitBarInterfaces from 'app/types/SubmitBar';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';

const mapStateToProps = (rootState: RootState) => {
  return {};
};

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    commitCode: (commitMessage: string) => dispatch(CodeActions.commit(commitMessage)),
    saveCode: () => dispatch(CodeActions.save()),
  };
};

const submitBarContainer = connect<{}, SubmitBarInterfaces.DispatchProps, {}>(
  mapStateToProps,
  mapDispatchToProps,
)(SubmitBar);

export default submitBarContainer;
