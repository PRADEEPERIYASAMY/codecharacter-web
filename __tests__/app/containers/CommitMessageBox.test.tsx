import { CommitMessageBox } from 'app/components/SubmitBar/CommitMessageBox';
import { mount } from 'enzyme';
import * as React from 'react';

describe('CommitMessageBox Container', () => {
  const props = {
    commitMessage: 'Inital Commit',
    handleCommit: () => {
      return;
    },
    isCommitMessageBoxOpen: true,
    updateCommitMessage: (commitMessage: string) => {
      return;
    },
  };
  const wrapper = mount(<CommitMessageBox {...props} />);
  it('Should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
