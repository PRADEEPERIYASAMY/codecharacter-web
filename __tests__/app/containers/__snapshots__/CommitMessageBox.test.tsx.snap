// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitMessageBox Container Should render 1`] = `
<CommitMessageBox
  commitMessage="Inital Commit"
  handleCommit={[Function]}
  isCommitMessageBoxOpen={true}
  updateCommitMessage={[Function]}
>
  <div
    className=""
  >
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <Grid
        bsClass="container"
        className=""
        componentClass="div"
        fluid={true}
      >
        <div
          className="container-fluid"
        >
          <Row
            bsClass="row"
            className="mt-2"
            componentClass="div"
          >
            <div
              className="mt-2 row"
            >
              <div
                className="w-100"
              >
                <div
                  className="col form-row mb-2"
                >
                  <div
                    className="input-group"
                  >
                    <input
                      className="form-control"
                      id="validationUsername"
                      maxLength={50}
                      onChange={[Function]}
                      placeholder="Describe your commit."
                      required={true}
                      value="Inital Commit"
                    />
                    <button
                      className="btn btn-info ml-3"
                      type="submit"
                    >
                      Commit
                    </button>
                    <div
                      className="invalid-feedback"
                    >
                      Please enter valid commit message.
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </Row>
        </div>
      </Grid>
    </form>
  </div>
</CommitMessageBox>
`;
